### ISSUE TITLE
diff function returns negative zero (-0) 

### ISSUE THREAD

**Issue by liesislukas** [moment/moment#3227]
in theoretical math you can have negative zero while representing movement around zero. In computer science you can also have negative zeros when negative quantity [underflows to zero](http://www.johndcook.com/blog/2010/06/15/why-computers-have-signed-zero/)

Should diff function return integer w/o negative zero?

code:

```
    let days_left = moment(this.props.task.get('due_date')).diff(moment(), 'days');
    console.log('days_left: ', this.props.task.get('due_date'), days_left, moment().toDate().toISOString());
```

console output:

```
days_left:  2016-06-08T21:00:00.000Z -1 2016-06-10T08:12:06.820Z
days_left:  2016-06-09T21:00:00.000Z -0 2016-06-10T08:12:06.845Z
days_left:  2016-06-10T21:00:00.000Z 0 2016-06-10T08:12:06.869Z
```

to have better visual, screenshot of console:
![image](https://cloud.githubusercontent.com/assets/2733862/15958118/488fcffe-2efc-11e6-8461-6b05ad27dd2a.png)

**Comment by maggiepint** [comment]
Runnable code to reproduce:

```js
moment('2016-06-09T21:00:00.00Z').diff('2016-06-10T08:12:06.845Z', 'days')
```

As a general rule, we don't want anything returning negative zero. Computer scientists have their reasons for making floating point as they do. Our users though, mostly just want to format the difference between dates in a way that makes sense to their users.

**Comment by maggiepint** [comment]
closed in favor of #3232
